remote-api/src/test/java/org/alfresco/rest/api/impl/RulesImplTest.java

```diff
@@ -281,6 +284,115 @@ public void testGetRuleByIdForNotAssociatedRuleToRuleSet()
+        given(ruleServiceMock.saveRule(folderNodeRef, serviceRuleBody)).willReturn(serviceRule);
+        given(serviceRule.getNodeRef()).willReturn(ruleNodeRef);
+
+        // when
+        List<Rule> actual = rules.saveRules(folderNodeRef.getId(), ruleSetNodeRef.getId(), ruleList);
+
+        then(ruleServiceMock).should().isRuleSetAssociatedWithFolder(ruleSetNodeRef, folderNodeRef);
+        then(ruleServiceMock).should().saveRule(folderNodeRef, ruleBody.toServiceModel(nodesMock));
+        then(ruleServiceMock).shouldHaveNoMoreInteractions();
+        List<Rule> expected = List.of(Rule.from(serviceRule));
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    /** Check that when passing the default rule set then we don't perform any validation around the rule set node. */
+    @Test
+    public void testSaveRules_defaultRuleSet()
+    {
+        NodeRef defaultRuleSetNodeRef = new NodeRef("default://rule/set");
+        given(ruleServiceMock.getRuleSetNode(folderNodeRef)).willReturn(defaultRuleSetNodeRef);
+        Rule ruleBody = mock(Rule.class);
+        List<Rule> ruleList = List.of(ruleBody);
+        org.alfresco.service.cmr.rule.Rule serviceRuleBody = mock(org.alfresco.service.cmr.rule.Rule.class);
+        given(ruleBody.toServiceModel(nodesMock)).willReturn(serviceRuleBody);
+        org.alfresco.service.cmr.rule.Rule serviceRule = mock(org.alfresco.service.cmr.rule.Rule.class);
+        given(ruleServiceMock.saveRule(folderNodeRef, serviceRuleBody)).willReturn(serviceRule);
+        given(serviceRule.getNodeRef()).willReturn(ruleNodeRef);
+
+        // when
+        List<Rule> actual = rules.saveRules(folderNodeRef.getId(), DEFAULT_ID, ruleList);
+
+        then(ruleServiceMock).should().getRuleSetNode(folderNodeRef);
+        then(ruleServiceMock).should().saveRule(folderNodeRef, ruleBody.toServiceModel(nodesMock));
+        then(ruleServiceMock).shouldHaveNoMoreInteractions();
+        List<Rule> expected = List.of(Rule.from(serviceRule));
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    public void testSaveRules_ruleSetNotAssociatedWithFolder()
+    {
+        Rule rule = new RuleBuilder().setName(RULE_NAME)
+                                     .createRule();
+        List<Rule> ruleList = List.of(rule);
+        given(ruleServiceMock.isRuleSetAssociatedWithFolder(ruleSetNodeRef, folderNodeRef)).willReturn(false);
+
+        // when
+        assertThatExceptionOfType(InvalidArgumentException.class).isThrownBy(
+                () -> rules.saveRules(folderNodeRef.getId(), ruleSetNodeRef.getId(), ruleList));
+
+        then(ruleServiceMock).should().isRuleSetAssociatedWithFolder(ruleSetNodeRef, folderNodeRef);
+        then(ruleServiceMock).shouldHaveNoMoreInteractions();
+    }
+
+    @Test
+    public void testSaveRules_emptyRuleList()
+    {
+        given(ruleServiceMock.isRuleSetAssociatedWithFolder(any(), any())).willReturn(true);
+        List<Rule> ruleList = List.of();
```

Comment:
Again - only opinion - for me `Collections.emptyList()` is more readable here but functionally it doesn't make too much difference anymore.